islandora_compound_object.module

```diff
@@ -268,17 +268,67 @@ function islandora_compound_object_access(AbstractObject $object) {
 }
 
 /**
- * Replaces parent object thumbnail with child's thumbnail.
+ * Implements hook_compound_cmodel_islandora_derivative().
  */
-function islandora_compound_object_update_parent_thumbnail($parent) {
+function islandora_compound_object_compoundCModel_islandora_derivative(AbstractObject $object = NULL, $ds_modified_params = array()) {
+  $module_path = drupal_get_path('module', 'islandora_compound_object');
+  $derivatives = array();
+  if (variable_get('islandora_compound_object_tn_deriv_hooks', FALSE)) {
+    $derivatives[] = array(
+      'source_dsid' => NULL,
+      'destination_dsid' => 'TN',
+      'function' => array('islandora_compound_object_create_thumbnail'),
+      'file' => "$module_path/includes/derivatives.inc",
+    );
+  }
+
+  return $derivatives;
+
+}
+
+/**
+ * Implements hook_islandora_derivative().
+ */
+function islandora_compound_object_islandora_derivative(AbstractObject $object) {
```

Comment:
Can we please, please respect function signatures?
https://github.com/Islandora/islandora/blob/83a0781e7011f300bbec9cee876b0efaa405e9e2/islandora.api.php#L706 
This here breaks any attempt to simply invoke the hooks to know if something is available or not at all but mostly, it is a bad practice  to override expected behaviour of hooks's signature. Makes coding unreliable